### PR TITLE
fix!: replace outdated webpush dependency

### PR DESCRIPTION
**Pull request by ZheSun88**
if flow build is green, let us take this method instead of #17940

### CODE REVIEW

File: flow-webpush/src/main/java/com/vaadin/flow/server/webpush/WebPush.java [hunk 1]
```diff
@@ -113,8 +115,10 @@ public WebPush(String publicKey, String privateKey, String subject) {
     public void sendNotification(Subscription subscription,
             WebPushMessage message) throws WebPushException {
         try {
-            HttpResponse response = pushService
-                    .send(new Notification(subscription, message.toJson()));
+            ;
```
Comment (mshabarov) [inline flow-webpush/src/main/java/com/vaadin/flow/server/webpush/WebPush.java:118]:
```suggestion
```

File: flow-webpush/pom.xml [hunk 2]
```diff
@@ -28,24 +28,17 @@
...
+            <artifactId>webpush-jwt-jose4j</artifactId>
+            <version>6.1.1</version>
         </dependency>
-
         <dependency>
-            <groupId>org.bouncycastle</groupId>
-            <artifactId>bcprov-jdk15on</artifactId>
-            <version>1.70</version>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpcore</artifactId>
```
Comment (mshabarov) [inline flow-webpush/pom.xml:37]: This apparently causes the following error when the push notification is being sent.
I got this error with CRM tutorial example:

```
java.lang.ClassCastException: class jdk.internal.net.http.HttpResponseImpl cannot be cast to class org.apache.http.HttpResponse (jdk.internal.net.http.HttpResponseImpl is in module java.net.http of loader 'platform'; org.apache.http.HttpResponse is in unnamed module of loader 'app')
        at com.vaadin.flow.server.webpush.WebPush.sendNotification(WebPush.java:120) ~[flow-webpush-24.3-SNAPSHOT.jar:na]
        at com.example.application.views.MessageView.lambda$getMessageBuilder$0(MessageView.java:57) ~[classes/:na]
        at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]

```

Comment (ZheSun88) [reply flow-webpush/pom.xml:37]: probably the implementation here needs to be updated. https://github.com/vaadin/flow/blob/7cc38fa79dace34aa47dd86bc1ff5e83eb976d47/flow-webpush/src/main/java/com/vaadin/flow/server/webpush/WebPush.java#L118

there is a cast...

Comment (mshabarov) [reply flow-webpush/pom.xml:37]: Let me try to replace it to `jdk.internal.net.http.HttpResponse`, it needs a couple of tweaks in code.

Comment (ZheSun88) [reply flow-webpush/pom.xml:37]: Thanks

**Review by mshabarov — APPROVED** [review]
Tested with https://github.com/vaadin/flow-crm-tutorial/tree/feature/webpush , works as expected. Many thanks to @ZheSun88 and @manolo 🥇

**Comment by mshabarov** [comment]
This fork has changes in `Subscription` API: things like `subscription.endpoint` -> `subscription.endpoint()`.
Easy to fix however, so I propose to pick this change into `24.2` series.

**Pull request merged**